src/battle_script_commands.c

```diff
@@ -5395,7 +5570,7 @@ static void Cmd_moveend(void)
                     BtlController_EmitSetMonData(0, REQUEST_STATUS_BATTLE, 0, 4, &gBattleMons[gActiveBattler].status1);
                     MarkBattlerForControllerExec(gActiveBattler);
                     effect = TRUE;
-                    BattleScriptPush(gBattlescriptCurrInstr + 1);
+                    BattleScriptPush(gBattlescriptCurrInstr);
```

Comment:
Note: Like the above, `gBattlescriptCurrInstr + 1` is inside the `moveend` instruction.

I imagine you can test this works with the same Life Orb thing, except that you need to cause a status to be cured by the move instead of Smacking Down a non-groupnded 'mon.